custom/ewsghana/api.py

```diff
@@ -145,6 +147,9 @@ def _create_location_from_supply_point(self, supply_point, location):
             if supply_point.supervised_by:
                 new_location.metadata['supervised_by'] = supply_point.supervised_by
             new_location.save()
+            sql_loc = new_location.sql_location
+            sql_loc.products = SQLProduct.objects.filter(code__in=supply_point.products)
+            sql_loc.save()
```

Comment:
This part of code is only called once per supply point / location. Only if location doesn't exist.